src/snowflake/connector/result_batch.py

```diff
@@ -437,10 +435,8 @@ def _load(self, response: "Response") -> List:
             Unfortunately there's not type hint for this.
             For context: https://github.com/python/typing/issues/182
         """
-        with GzipFile(fileobj=response.raw, mode="r") as gfd:
-            read_data: str = gfd.read().decode("utf-8", "replace")
-            self._done_load(response)
-            return json.loads("".join(["[", read_data, "]"]))
+        read_data = response.text
```

Comment:
`response.text` gives us the response as a string and it automatically decompresses gzip content, so no reading, decoding, or decompressing is necessary